paperscraper/utils.py

```diff
@@ -104,22 +98,20 @@ async def _allow(self) -> None:
 
     async def _request(self, *args, **kwargs) -> aiohttp.ClientResponse:
         """Throttled _request()."""
-        for retries in range(5):
-            await self._allow()
+        for retry_num in range(5):
+            await self._wait_can_make_request()
             response = await super()._request(*args, **kwargs)
-            if response.status in self.SERVICE_LIMIT_REACHED_STATUS_CODES:
-                if self.rate_limit is not None:
-                    await asyncio.sleep(
-                        max(
-                            3 * self.rate_limit,
-                            (2**retries) * 0.1 + random.random() * 0.1,
-                        )
-                    )
-                    continue
-                raise NotImplementedError(
-                    "Hit a service limit without a rate limit, please specify a rate limit."
-                )
-            break
+            if response.status not in self.SERVICE_LIMIT_REACHED_STATUS_CODES:
+                break
+            if self._rate_limit is not None:
```

Comment:
Can we optionally pass a logger into this object, and debug log when this happening? I have had a circumstance where my only recourse is to step through the whole stack to ensure that the rate limits are where I'm hanging.

Reply:
Imo if `rate_limit` is `None`, we should just use `aiohttp.ClientSession` directly. What do you think of disallowing a `None` `rate_limit` for this subclass, with an appropriate warning?

Reply:
I think that's fine, but on the logging front I'd still like to know if it's in a rate-limit backoff loop as part of the debug logs.

Reply:
I am just gonna close this an open a separate one

Reply:
I decided not to deny `rate_limit` of `None` because I wasn't sure this would annoy us in the future